File: src/main/java/de/fhswf/genericapplication/models/core/PDAssociationMeta.java
package de.fhswf.genericapplication.models.core;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.xml.bind.annotation.*;
import java.util.ArrayList;
import java.util.List;

@XmlRootElement(namespace = "core", name = "association")
@XmlAccessorType(XmlAccessType.FIELD)
public class PDAssociationMeta extends PDMemberMeta {
    @XmlAttribute
    @JsonProperty
    private String type;

    @XmlAttribute
    @JsonProperty
    private boolean required = false;

    @XmlAttribute
    @JsonProperty
    private String inverseOf;

    @XmlElementWrapper(namespace = "core", name = "validators")
    @XmlElement(namespace = "core", name = "validator")
    @JsonProperty
    private List<ValidatorMeta> validators = new ArrayList<>();

    @XmlAttribute
    @JsonProperty
    private String cardinality;
}

File: src/main/ui/src/models/core/PDAssociationMeta.ts
import type {PDMemberMeta} from "@/models/core/PDMemberMeta";
import type {PDValidatorMeta} from "@/models/core/PDValidatorMeta";
import type {AssociationCardinalityET} from "@/models/runtime/AssociationCardinalityET";

export interface PDAssociationMeta extends PDMemberMeta {
    type: string;
    required: boolean;
    inverseOf: string;
    validators?: PDValidatorMeta[];
    cardinality?: AssociationCardinalityET;
}

export function isPDAssociationMeta(obj: any): obj is PDAssociationMeta {
    return 'required' in obj && 'type' in obj && 'inverseOf' in obj;
}
